lib/charms/opensearch/v0/opensearch_relation_provider.py

```diff
@@ -192,4 +193,4 @@ def update_endpoints(self, relation):
         """Updates endpoints in the databag for the given relation."""
         port = self.opensearch.port
         endpoints = [f"{ip}:{port}" for ip in units_ips(self.charm, PeerRelationName).values()]
-        self.database_provides.set_endpoints(relation.id, ",".join(endpoints))
+        self.opensearch_provides.set_endpoints(relation.id, ",".join(endpoints))
```

Comment:
please remember to update endpoints when there is a scale up/down

Reply:
Good point, I've added that in